src/main/webapp/feed.js

```diff
@@ -0,0 +1,468 @@
+    bookmarkNum.innerText = event.get("bookmarks");
+    return bookmarkNum;
+}
+
+function addEventUserText(event) {
+    const eventUser = document.createElement('p');
+    eventUser.className = "event-info";
+    eventUser.innerText = event.get("author") + " posted an event:";
+    return eventUser;
+}
+
+function addEventMiddleSection(event) {
+    const eventMiddle = document.createElement('div');
+    eventMiddle.className = "event-middle";
+    eventMiddle.appendChild(addEventInnerCard(event));
+    eventMiddle.appendChild(addEventDescription(event));
+    return eventMiddle;
+}
+
+function addEventInnerCard(event) {
+    const eventIcon = document.createElement('div');
+    eventIcon.className = "event-icon";
+    eventIcon.innerText = eventCategoryIcons.get(event.get("category"));
+
+    const eventTitle = document.createElement('p');
+    eventTitle.className = "event-title";
+    eventTitle.innerText = event.get("title");
+
+    const eventInnerCard = document.createElement('div');
+    eventInnerCard.className = "event-inner-card";
+    eventInnerCard.appendChild(eventIcon);
+    eventInnerCard.appendChild(eventTitle);
+    return eventInnerCard;
+}
+
+function addEventDescription(event) {
+    const eventDescription = document.createElement("p");
+    eventDescription.className = "event-description";
+    eventDescription.innerText = "description:\n\n" + event.get("description");
+    return eventDescription;
+}
+
+function addEventInfo(event) {
+    const eventInfo = document.createElement("div");
+    eventInfo.className = "event-info";
+
+    const eventLocation = document.createElement("p");
+    eventLocation.innerHTML =  "📍&nbsp&nbsp";
+    eventLocation.innerText += event.get("location");
+
+    const eventDate = document.createElement("p");
+    eventDate.innerHTML =  "📅&nbsp&nbsp";
+    eventDate.innerText += event.get("date");
+
+    eventInfo.appendChild(eventLocation);
+    eventInfo.appendChild(eventDate);
+    return eventInfo;
+}
+
+function setChallengeBox(challengeId) {
+    const badge = document.getElementById("challenges-badge");
+    const icon = document.getElementById("challenges-badge-icon");
+    const stepsText = document.getElementById("challenge-steps");
+
+    if (challengeId == -1) {
+        icon.innerText = "✔️";
+        stepsText.style.fontSize = "16px";
+        stepsText.innerText = "Complete!";
+    }
+    else {
+        icon.innerText = challenges[challengeId].get("icon");
+
+        const currentStep = user.get("challengeStatuses")[challengeId];
+        const totalSteps = challenges[challengeId].get("steps").length;
+        
+        stepsText.style.fontSize = "20px";
+        stepsText.innerText = currentStep + "/" + totalSteps;
+        
+        fillBadge(currentStep, totalSteps);
+    }
+}
+
+function fillBadge(currentStep, totalSteps) {
+    const badgeFilling = document.getElementById("feed-badge-filling")
+    badgeFilling.style.height = 120*(currentStep/totalSteps) + "px";
```

Comment:
What is the 120 from/for? Could this be a set constant if it is used multiple times